### PR TITLE
Configure native deepspeed schedulers with interval='step'

### LINKED ISSUE #11694: DeepSpeed learning rate scheduler not working
## 🐛 Bug

PyTorch Lightning does not appear to be using a learning rate scheduler specified in the DeepSpeed config as intended. It increments the learning rate only at the end of each epoch, rather than once per step. Additionally, it uses the wrong learning rate for the very first epoch.

### To Reproduce

I used the following DeepSpeed config:

```
{
  "optimizer": {
    "type": "Adam",
    "params": {
      "lr": 0.001,
      "eps": 1e-05
    }
  },
  "scheduler": {
      "type": "WarmupLR",
      "params": {
        "warmup_min_lr": 0,
        "warmup_max_lr": 0.001,
        "warmup_num_steps": 3000,
        "warmup_type": "linear"
      }
  },
  "fp16": {
    "enabled": false,
    "min_loss_scale": 1
  },
  "amp": {
    "enabled": false,
    "opt_level": "O2"
  },
  "bfloat16": {
    "enabled": true
  },
  "zero_optimization": {
    "stage": 2,
    "cpu_offload": false,
    "contiguous_gradients": false
  },
  "activation_checkpointing": {
    "partition_activations": true,
    "cpu_checkpointing": false,
    "profile": false
  },
  "gradient_clipping": 0.1
}
```

I then used a `pytorch_lightning.callbacks.lr_monitor.LearningRateMonitor` to monitor the learning rate. This is the result:

image

For the first epoch, the learning rate is 0.001. For the next one, it's zero. Thereafter, the learning rate is warmed up just once per epoch, rather than once per step.

It seems to me like the DeepSpeed LR is being updated on the "epoch" interval rather than the "step" one.

### Environment

* CUDA:
        - GPU:
        - available:         False
        - version:           11.3
* Packages:
        - numpy:             1.21.1
        - pyTorch_debug:     False
        - pyTorch_version:   1.10.1+cu113
        - pytorch-lightning: 1.5.9
        - tqdm:              4.62.2
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                -
        - processor:         x86_64
        - python:            3.7.0
        - version:           #1 SMP Wed Dec 1 05:10:33 EST 2021

cc @SeanNaren @awaelchli @rohitgr7

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fixes #11694

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @SeanNaren @awaelchli @rohitgr7

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]
amazing thanks @rohitgr7

**Review by tchaton — APPROVED** [review]
Neat !
